hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveSyncTool.java

```diff
@@ -106,13 +106,13 @@ public void syncHoodieTable() {
       if (hoodieHiveClient != null) {
         switch (hoodieHiveClient.getTableType()) {
           case COPY_ON_WRITE:
-            syncHoodieTable(snapshotTableName, false);
+            syncHoodieTable(snapshotTableName, false, false);
```

Comment:
I thought here we go for read optimized. do you know what happens if we do read optimized read on COW table in general?

Reply:
Hi @nsivabalan, we use the `getBaseFileOnlyView` to read data for COW for both read-optimized & snapshot in `DefaultSource`. So it does not matter here. we can see the code in `DefaultSource`:

> (tableType, queryType, isBootstrappedTable) match {
      case (COPY_ON_WRITE, QUERY_TYPE_SNAPSHOT_OPT_VAL, false) |
           (COPY_ON_WRITE, QUERY_TYPE_READ_OPTIMIZED_OPT_VAL, false) |
           (MERGE_ON_READ, QUERY_TYPE_READ_OPTIMIZED_OPT_VAL, false) =>
        getBaseFileOnlyView(useHoodieFileIndex, sqlContext, parameters, schema, tablePath,
          readPaths, metaClient)

Reply:
got it, thanks.